qiskit_quantuminspire/base_provider.py

```diff
@@ -0,0 +1,18 @@
+from abc import ABC, abstractmethod
+from typing import Optional, Sequence
+
+from qiskit.providers import BackendV2
+
+
+class BaseProvider(ABC):
```

Comment:
Having a interface to inherit from makes testing easier and avoid circular dependencies in QIJob.